sydent/util/versionstring.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import os
+import subprocess
+
+import sydent
+
+logger = logging.getLogger(__name__)
+
+
+def get_version_string() -> str:
+    """Calculate a git-aware version string for sydent.
+
+    Implementation nicked from Synapse.
```

Comment:
I think it's worth writing a bit more here as to what form the version string will actually take (given what information can be found).